packages/ui/react-ui-editor/src/extensions/tasklist.ts

```diff
@@ -2,118 +2,94 @@
 // Copyright 2023 DXOS.org
 //
 
-import {
-  EditorView,
-  Decoration,
-  WidgetType,
-  MatchDecorator,
-  ViewPlugin,
-  type DecorationSet,
-  type ViewUpdate,
-} from '@codemirror/view';
-
-// TODO(burdon): Reconcile with theme.
-const styles = EditorView.baseTheme({
-  '& .cm-task-item': {
-    paddingLeft: '4px',
-    paddingRight: '8px',
-  },
-});
+import { syntaxTree } from '@codemirror/language';
+import { RangeSetBuilder } from '@codemirror/state';
+import { EditorView, Decoration, WidgetType, ViewPlugin, type DecorationSet, type ViewUpdate } from '@codemirror/view';
 
 class CheckboxWidget extends WidgetType {
-  constructor(
-    private _checked: boolean,
-    private readonly _indent: number,
-    private readonly _onCheck?: (check: boolean) => void,
-  ) {
+  constructor(private _checked: boolean) {
     super();
   }
 
   override eq(other: this) {
-    return this._checked === other._checked && this._indent === other._indent;
+    return this._checked === other._checked;
   }
 
   override toDOM(view: EditorView) {
-    const wrap = document.createElement('span');
-    wrap.className = 'cm-task-item';
-    wrap.setAttribute('aria-hidden', 'true');
-    wrap.style.setProperty('margin-left', this._indent * 24 + 'px');
-
-    const input = wrap.appendChild(document.createElement('input'));
+    const input = document.createElement('input');
     input.type = 'checkbox';
     input.checked = this._checked;
-    if (!this._onCheck) {
+    if (view.state.readOnly) {
       input.setAttribute('disabled', 'true');
-    }
-    if (this._onCheck) {
-      input.onchange = (event: Event) => {
-        this._onCheck?.((event.target as any).checked);
-        return true;
+    } else {
+      input.onmousedown = (event: Event) => {
```

Comment:
curious why you chose this over onclick?
my understanding is onclick would also catch the case where the input is focused via the keyboard and then pressing enter?

Reply:
Once click (or change, or mouseup) fires, the focus has already been moved to the element, and lost the editor. Since I guess our keyboard store for these is going to be regular text editing, now that they expand when you move the cursor to them, I didn't really account for keyboard interaction.